### PR TITLE
Remove constant propagation expect files

### PR DESCRIPTION
**Pull request by eellison**
Remove constant prop expect files, and express graph conditions via python bindings. 

First diff in larger effort to remove expect files

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -1712,7 +1720,7 @@ def constant_prop():
             return b
 
         self.run_pass('constant_propagation', constant_prop.graph)
-        self.assertExpected(canonical(constant_prop.graph))
```
Comment (wanchaol) [inline test/test_jit.py]: what is this test really checking here? I think aten::randn will always be there without constant prop or not.

File: torch/csrc/jit/python_ir.cpp [hunk 2]
```diff
@@ -68,35 +68,55 @@ std::ostream& printPyObject(std::ostream& out, const THPObjectPtr& obj) {
   }
 }
 
-std::vector<Node*> findAllNodes(Block* block, Symbol kind) {
+std::vector<Node*> findAllNodes(
+    c10::ArrayRef<torch::jit::Block*> blocks,
+    Symbol kind,
+    bool recurse) {
```
Comment (wanchaol) [inline torch/csrc/jit/python_ir.cpp]: make `recurse` to have a default value in C++ as well?

**Review by wanchaol — APPROVED** [review]
lgtm, just have nits

File: test/test_jit.py (hunk 1 above)
Comment (eellison) [reply test/test_jit.py]: This was testing that randn was not run. DCE gets run after constant prop so if it was run it would no longer be in the graph

**Comment by suo** [comment]
Is this the eventual API you want clients to consume? If not, this seems like redundant work if you're going to have to migrate all these tests anyway.

**Comment by eellison** [comment]
@suo it depends on what you're testing. If you want to test properties of a node - like number of outputs here -  capturing that via regex is going to be hard & not very readable. I don't plan on migrating this test to regex - like testing.

**Comment by suo** [comment]
I more meant, should we be testing this through the python frontend at all? This seems like a good candidate for testing through an IR-to-IR framework—there's no point to running the frontend for a test like this.

**Comment by eellison** [comment]
@suo I agree that an optimization pass would be a good candidate for IR testing, however this removes expect files & is therefore an improvement. I don't know when IR->IR parsing will be usable so I would prefer to land this.

File: test/test_jit.py [hunk 3]
```diff
@@ -1783,8 +1791,17 @@ def constant_prop(a, b):
                 c2 = c2 + 4  # set to 5
             return a + c0 + c1 + c2
 
-        self.run_pass('constant_propagation', constant_prop.graph)
-        self.assertExpected(canonical(constant_prop.graph))
+        graph = constant_prop.graph
+        self.run_pass('constant_propagation', graph)
+        recurse = False
```
Comment (apaszke) [inline test/test_jit.py]: nit: would be somewhat cleaner to say `recurse=False` when you call `find*`

**Review by apaszke — APPROVED** [review]

**Pull request closed without merging**